### ISSUE TITLE
Tmux resurrect file not found!

### ISSUE THREAD

**Issue by janzenz** [tmux-plugins/tmux-continuum#33]
I get this error when running tmux.

Running on OSX Sierra 10.12.2
Tmux 2.3

Here is my tmux.conf file:

```
# TMUX Ressurect plugin
set -g @plugin 'tmux-plugins/tpm'
set -g @plugin 'tmux-plugins/tmux-resurrect'
set -g @plugin 'tmux-plugins/tmux-continuum'

set -g @continuum-save-interval '5'
set -g @continuum-restore 'on'

run '~/.tmux/plugins/tpm/tpm'
```

**Comment by janzenz** [comment]
The `resurrect` directory did exist but looks like it was caused by the lack of saved session in the directory. This had me confused and thought I installed the plugin incorrectly. For newbs like me I think it would help if the error would be more descriptive.

**Comment by omenius** [comment]
I had same error, after trying to save session. chmod 755 ~/.tmux fixed it. Problem was that it was never saving the session because of permission issues.

**Comment by jimafisk** [comment]
Yeah.... and this is obvious, but make sure you've actually saved a session (`prefix` + <kbd>ctrl</kbd>-<kbd>s</kbd>) before trying to restore (`prefix` + <kbd>ctrl</kbd>-<kbd>r</kbd>). :man_facepalming:

**Comment by forkjoseph** [comment]
I had a same issue. Resolved the issue by 
1) ls -lah ~/.tmux/tmux-resurrect/last
2) cd ~/ && ls -lah .

if your tmux folder is owned by "root", not username like this
`drwxr-xr-x  3 root      root      4.0K Jul 21 17:33 .tmux/    `
Make sure to change the ownership by 
`sudo chown $(whoami):$(whoami) -R .tmux `

My stupidity of using root privilege whilst installing resurrect caused this issue....

**Comment by babenzele** [comment]
This error occurred for me when the symbolic link `last` was pointing to a non-existent resurrect file. I'm guessing I quit tmux while tmux-resurrect was saving.

To solve I created a new link pointing to the last created save:

`ln -s tmux_resurrect_2017-09-08T09:53:43.txt last`

**Comment by cabaalexander** [comment]
If you try to save it from a vagrant machine into the host via shared folder it wont work because of the permissions as @omenius said :'/

**Comment by Blake-LeBlanc** [comment]
Thank you @tim-phillips, your solution is what did the trick on my end!

**Comment by valentinrusu-movidius** [comment]
This happened to me just after restarting my computer using the "reboot" command line. I suppose it somehow caught the resurrect plugin in the middle of saving the session and it never got the chance to create the "last" symbolic link before the shutdown. Thank you @tim-phillips for the tip.

**Comment by reorx** [comment]
I wrote a shell function to help dealing with this:

```
function tmux-resurrect-reset-last() {
    cd ~/.tmux/resurrect && \
        ln -f -s $(/bin/ls -t tmux_resurrect_*.txt | head -n 1) last && \
        /bin/ls -l last
}
```

**Comment by valentinrusu-movidius** [comment]
This issue no longer happens on latest updates.

**Comment by dreamflasher** [comment]
Still happening for me, latest version. I regularly come to this issue and try a couple of the solutions but never manage to find out which one actually does the trick.

**Comment by Bogdan-Torkhov** [comment]
> Yeah.... and this is obvious, but make sure you've actually saved a session (`prefix` + ctrl-s) before trying to restore (`prefix` + ctrl-r). man_facepalming

These is the solution!

**Comment by flakrat** [comment]
Happened to me on a Mac today after rebooting. After a lot testing, I finally decided to change the following in my `~/.tmux.conf.local` file

- From

```shell
set -g @plugin 'tmux-plugins/tmux-resurrect'
set -g @resurrect-dir '$HOME/.tmux_resurrect'
```

To hardcoding the path

```shell
set -g @plugin 'tmux-plugins/tmux-resurrect'
set -g @resurrect-dir '/Users/myuserid/.tmux_resurrect'
```

No idea why using `$HOME` stopped working as I didn't change anything and I've used that in `resurrect-dir` for years. I use `$HOME` since I clone my dotfiles across many Linux and Mac systems, and the home directory / userid vary depending on the system.

**Comment by ivanp7** [comment]
> No idea why using `$HOME` stopped working as I didn't change anything and I've used that in `resurrect-dir` for years. I use `$HOME` since I clone my dotfiles across many Linux and Mac systems, and the home directory / userid vary depending on the system.

Had the same problem. Tried to use `~` instead of `$HOME`, successfully.

**Comment by m4c0** [comment]
This one is horrid for troubleshooting. I’m not sure why the report is in this repo because the error message is in `tmux-continuum`.

The error is triggered by a `[ -f … ]` criteria. If you know shell, you know that error can mean a lot of stuff - including missing directory, etc.

In my case it was misspelling of `resurrect` - not fun when English is not your first language… 😐

**Comment by mateusz-** [comment]
On Windows with MSYS2, I believe creating symlinks with `MSYS=winsymlinks:nativestrict` environment variable requires admin privileges. In order to get around this issue `MSYS=winsymlinks:native` can be used.

Basically, Windows will fail to create the `last` tmux symlink.

**Comment by sxhmilyoyo** [comment]
I have to manually create the directory `~/.tmux/resurrect` to make it work.
